lib/pbench/agent/tool_data_sink.py

```
@@ -251,48 +290,56 @@ def launch(self):
                 f"{self.benchmark_run_dir}/tm/prometheus.yml:/etc/prometheus/prometheus.yml:Z",
                 "--network",
                 "host",
-                prom_reg,
+                self.prom_reg,
             ]
             try:
                 self.run = subprocess.Popen(args, stdout=prom_logs, stderr=prom_logs)
             except Exception as exc:
-                self.logger.error("Podman run process failed: '%s'", exc)
+                self.logger.error("Podman run process failed: '%s', %r", exc, args)
                 self.run = None
-                return 0
-        return 1
 
     def terminate(self):
-        if super().terminate() == 0:
-            return 0
+        """terminate - shuts down the prometheus sub-process, and creates a
+        tar ball of the data collected.
+        """
+        try:
+            super().terminate()
+        except Exception:
+            self.logger.error("Prometheus failed to terminate")
+            return
 
         self.logger.debug("Prometheus terminated")
 
         args = [
-            "tar",
+            tar_path,
             "--remove-files",
-            "--exclude",
-            "prometheus/prometheus_data.tar.gz",
-            "-zcvf",
-            f"{self.volume}/prometheus_data.tar.gz",
+            "-zcf",
+            f"{self.tool_group_dir}/prometheus_data.tar.gz",
             "-C",
             f"{self.tool_group_dir}/",
             "prometheus",
         ]
         data_store = subprocess.Popen(args)
         data_store.wait()
```

Consider hiding exceptions encountered tar'ing up the data.

---

Tracked in https://github.com/distributed-system-analysis/pbench/issues/2161.